app/models/question.rb

```diff
@@ -36,4 +38,8 @@ def total_rates
   def like_ratio
     total_rates == 0 ? 0.to_f : positive_rates.to_f / total_rates
   end
+
+  def convert_correct_answers
```

Comment:
On the new and edit views because of the tricky checkbox use for correct_answers the value returned are indexes corresponding to the answers attribute.
The params sent by the form are therefore
answers: ['reponse1', 'reponse2', 'reponse3', 'reponse4']
correct_answers: ['1', '3']

the method convert_correct_answers is called with the callback before_save before a create or a update so that it become correct_answers: ['reponse1', 'reponse3']